hsm/m0hsm_api.c

```diff
@@ -2057,7 +2062,8 @@ int m0hsm_test_read(struct m0_uint128 id, off_t offset, size_t len)
 	memset(&obj, 0, sizeof(struct m0_obj));
 
 	/* Set the object entity we want to write */
-	m0_obj_init(&obj, m0_uber_realm, &id, m0_client_layout_id(m0_instance));
+	m0_obj_init(&obj, m0_uber_realm, &id,
+		    m0_client_layout_id(m0_instance), 0);
```

Comment:
m0_obj already has object size field, so we can avoid passing object size as separate param.

Reply:
My thinking was to use the already present initializer function `m0_obj_init()` which already populates the `struct m0_obj` and `struct m0_obj_attr`, so if we are adding a new field in `struct m0_obj_attr`  it seems more appropriate to use the initializer function to set the new field as well.
@madhavemuri  what are your thoughts on this?

Reply:
I agree not to change the public API: m0_obj_init().
This API is publicly exported in "motr/client.h".
Changing this API will affect a lot of other application, e.g. S3 and other.
We need to keep all the Motr APIs as stable as possible.

Reply:
Done